ui-components/src/headers/SiteHeader.tsx

```diff
@@ -30,61 +33,73 @@ export const NavbarDropdown = (props: NavbarDropdownProps) => {
   return (
     <div className="navbar-item has-dropdown is-hoverable" tabIndex={0}>
       <a className="navbar-link">{props.menuTitle}</a>
-
       <div className="navbar-dropdown">{props.children}</div>
     </div>
   )
 }
 
-class SiteHeader extends React.Component<SiteHeaderProps, SiteHeaderState> {
-  constructor(props: SiteHeaderProps) {
-    super(props)
-    this.state = {
-      active: false,
-    }
-    this.handleMenuToggle = this.handleMenuToggle.bind(this)
-  }
+const SiteHeader = (props: SiteHeaderProps) => {
+  const [active, setActive] = useState(false)
 
-  skipLink() {
+  const skipLink = () => {
     return (
       <a href="#main-content" className="navbar__skip-link">
-        {this.props.skip}
+        {props.skip}
       </a>
     )
   }
 
-  noticeBar() {
+  const noticeBar = () => {
     return (
       <div className="navbar-notice">
-        <div className="navbar-notice__text">{this.props.notice}</div>
+        <div className="navbar-notice__text">{props.notice}</div>
       </div>
     )
   }
 
-  logo(logoClass = "") {
+  const getLogoWidthClass = () => {
+    if (props.logoWidth === "slim") return "navbar-width-slim"
+    if (props.logoWidth === "medium") return "navbar-width-med"
+    if (props.logoWidth === "wide") return "navbar-width-wide"
+    return ""
+  }
```

Comment:
This is a preference of mine, that doesn't need to be changed, but in cases like this, I like switch cases.